webapp/loaders.py

```diff
@@ -71,13 +52,24 @@ def find_template(self, name, dirs=None):
         raise TemplateDoesNotExist(name)
 
     def load_template_source(self, template_name, template_dirs=None):
+        """
+        Look for template through loaders, including checking if the source
+        is a folder with an index file.
+        """
+        template_name_parts = template_name.split('.md')
+        template_name_index = ''.join([template_name_parts[0], '/index.md'])
+        template_names = [
+            template_name,
+            template_name_index,
+        ]
         for loader in self.loaders:
-            try:
-                return loader.load_template_source(
-                    template_name, template_dirs
-                )
-            except TemplateDoesNotExist:
-                pass
+            for template_name in template_names:
+                try:
+                    return loader.load_template_source(
+                        template_name, template_dirs
+                    )
+                except TemplateDoesNotExist:
```

Comment:
OK, I remember now after sleep!
We are looping over the loaders and requesting template names. It expects it to return this exception sometimes.

So, the first loader may return the exception, then the second one finds the template.